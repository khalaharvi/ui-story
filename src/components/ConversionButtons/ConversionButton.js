import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import { cx } from './ConversionButtons';

const ConversionButton = ({
  icon,
  type,
  onClick,
  label,
  url,
  attributes
}) => (
  <Fragment>
    {
      (url) ?
        <a
          href={url}
          className={classnames(cx('btn'), cx(`btn-${type}`))}
          onClick={onClick}
          {...attributes}
        >
          {icon}
        </a> :
        <button
          className={classnames(cx('btn'), cx(`btn-${type}`))}
          type="button"
          onClick={onClick}
          aria-label={label}
          {...attributes}
        >
          {icon}
        </button>
    }
  </Fragment>
);

ConversionButton.propTypes = {
  icon: PropTypes.element.isRequired,
  type: PropTypes.oneOf(['facebook', 'twitter', 'pinterest', 'email', 'print', 'download']).isRequired,
  onClick: PropTypes.func,
  label: PropTypes.string,
  url: PropTypes.string,
  attributes: PropTypes.object // eslint-disable-line react/forbid-prop-types
};

export default ConversionButton;
