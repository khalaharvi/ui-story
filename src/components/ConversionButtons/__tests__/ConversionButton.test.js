import React from 'react';
import { shallow } from 'enzyme';

import { Facebook as FacebookIcon } from 'icons';
import ConversionButton from '../ConversionButton';

describe('<ConversionButton />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <ConversionButton
        icon={<FacebookIcon />}
        type="facebook"
      />
    );

    expect(wrapper.find('button')).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with a label', () => {
    const wrapper = shallow(
      <ConversionButton
        icon={<FacebookIcon />}
        type="facebook"
        label="Share on Facebook"
      />
    );

    expect(wrapper.find('[aria-label]')).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with attributes', () => {
    const wrapper = shallow(
      <ConversionButton
        icon={<FacebookIcon />}
        type="facebook"
        label="Share on Facebook"
        attributes={{
          'data-test': 'test'
        }}
      />
    );

    expect(wrapper.find('[data-test]')).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly has a link', () => {
    const wrapper = shallow(
      <ConversionButton
        icon={<FacebookIcon />}
        type="facebook"
        url="#"
      />
    );

    expect(wrapper.find('a')).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call onClick when clicked', () => {
    const onClick = jest.fn();
    const wrapper = shallow(
      <ConversionButton
        icon={<FacebookIcon />}
        type="facebook"
        onClick={onClick}
      />
    );

    wrapper.find('button').simulate('click');
    expect(onClick.mock.calls).toHaveLength(1);
  });
});
