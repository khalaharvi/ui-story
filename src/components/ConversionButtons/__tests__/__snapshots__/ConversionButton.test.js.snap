// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConversionButton /> renders correctly 1`] = `
<React.Fragment>
  <button
    className="dh-conversion-buttons-btn dh-conversion-buttons-btn-facebook"
    type="button"
  >
    <Facebook />
  </button>
</React.Fragment>
`;

exports[`<ConversionButton /> renders correctly has a link 1`] = `
<React.Fragment>
  <a
    className="dh-conversion-buttons-btn dh-conversion-buttons-btn-facebook"
    href="#"
  >
    <Facebook />
  </a>
</React.Fragment>
`;

exports[`<ConversionButton /> renders correctly with a label 1`] = `
<React.Fragment>
  <button
    aria-label="Share on Facebook"
    className="dh-conversion-buttons-btn dh-conversion-buttons-btn-facebook"
    type="button"
  >
    <Facebook />
  </button>
</React.Fragment>
`;

exports[`<ConversionButton /> renders correctly with attributes 1`] = `
<React.Fragment>
  <button
    aria-label="Share on Facebook"
    className="dh-conversion-buttons-btn dh-conversion-buttons-btn-facebook"
    data-test="test"
    type="button"
  >
    <Facebook />
  </button>
</React.Fragment>
`;
